libs/ui-lib/lib/ocm/components/clusterWizard/wizardTransition.test.ts

```diff
@@ -0,0 +1,472 @@
+  infrastructure: [
+    {
+      id: 'container-images-available',
+      message: 'a dummy host validation unrelated to the canNextHostDiscovery',
+      status: 'error',
+    },
+  ],
+  hardware: [
+    {
+      id: 'hostname-unique',
+      message: 'a validation message',
+      status: 'success',
+    },
+    {
+      id: 'hostname-valid',
+      message: 'a validation message',
+      status: 'pending',
+    },
+  ],
+};
+const hostValidationInfoStorage = hostValidationInfoHostsDiscovery;
+
+const hostValidationInfoNetwork: HostValidationsInfo = {
+  infrastructure: [
+    {
+      id: 'container-images-available',
+      message: 'a dummy host validation unrelated to the canNextHostDiscovery',
+      status: 'error',
+    },
+  ],
+  network: [
+    {
+      id: 'ntp-synced',
+      message: 'a soft validation message',
+      status: 'pending',
+    },
+    {
+      id: 'api-domain-name-resolved-correctly',
+      message: 'a validation message',
+      status: 'pending',
+    },
+    {
+      id: 'belongs-to-majority-group',
+      message: 'a validation message',
+      status: 'success',
+    },
+  ],
+};
+
+// For simplicity, let's put all required validations in a flat structure
+const flatValidationHostsDiscovery: HostValidation[] = [
+  {
+    id: 'connected',
+    message: 'a host validation message',
+    status: 'success',
+  },
+  {
+    id: 'media-connected',
+    message: 'a host validation message',
+    status: 'success',
+  },
+  {
+    id: 'lso-requirements-satisfied',
+    message: 'a host validation message',
+    status: 'success',
+  },
+  {
+    id: 'odf-requirements-satisfied',
+    message: 'a host validation message',
+    status: 'success',
+  },
+  {
+    id: 'lvm-requirements-satisfied',
+    message: 'a host validation message',
+    status: 'success',
+  },
+  {
+    id: 'cnv-requirements-satisfied',
+    message: 'a host validation message',
+    status: 'success',
+  },
+];
+
+const flatValidationHostsStorage: HostValidation[] = [
+  {
+    id: 'connected',
+    message: 'a host validation message',
+    status: 'success',
+  },
+  {
+    id: 'media-connected',
+    message: 'a host validation message',
+    status: 'success',
+  },
+  {
+    id: 'no-skip-installation-disk',
+    message: 'a host validation message',
+    status: 'success',
+  },
+  {
+    id: 'no-skip-missing-disk',
+    message: 'a host validation message',
+    status: 'success',
+  },
+];
+
+const flatValidationHostsNetwork: HostValidation[] = [];
+
+describe('OCM wizard transitions', () => {
+  test('canNextClusterDetails', () => {
+    // based on clusterDetailsStepValidationsMap
+    const cluster: Cluster = cloneDeep(clusterBase);
+    const validationInfo = cloneDeep(validationInfoClusterDetails);
+
+    expect(canNextClusterDetails({ cluster })).toBe(false);
+
+    cluster.status = 'ready';
+    expect(canNextClusterDetails({ cluster })).toBe(true);
+
+    cluster.status = clusterBase.status;
```

Comment:
I would rather extract each case into its own "test" unit.

This would allow:
 - To have a self-explanatory title of what each test is verifying
 - To make sure each test is independent (you could have a `beforeEach` that defines the initial cluster and validation info, therefore nothing from a previous test can affect a subsequent test)
 - To be able to see at a glance what's being covered in each "describe" block and what may be missing. Currently, you would need to read the entire test to understand that.

Reply:
I believe this is done. Resolving